app/src/main/java/ingsw/group1/findmyphone/contacts/SMSContactManager.java

```diff
@@ -0,0 +1,181 @@
+
+    /**
+     * Private constructor
+     *
+     * @param applicationContext    {@link Context} of the application
+     *
+     * @throws ExceptionInInitializerError if the object is already initialized
+     */
+    private SMSContactManager(@NonNull Context applicationContext) throws ExceptionInInitializerError {
+        if(instance != null)
+            throw new ExceptionInInitializerError();
+
+        contactDatabase = Room.databaseBuilder(applicationContext, SMSContactDatabase.class,
+                CONTACTS_DB_NAME)
+                .enableMultiInstanceInvalidation()
+                .allowMainThreadQueries()
+                .build();
+    }
+
+    /**
+     * Method to get the only valid on-disk instance of this class. A new instance is created only if it was
+     * null previously. The used context is always the parent application context of the parameter.
+     *
+     * @param applicationContext    The calling context
+     *
+     * @return the only instance of {@link SMSContactManager}
+     */
+    public static SMSContactManager getInstance(@NonNull Context applicationContext) {
+        if (instance == null) {
+            instance = new SMSContactManager(applicationContext);
+        }
+        return instance;
+    }
+
+    //---------------------------- OPERATIONS ON THE SINGLE CONTACT ----------------------------
+
+    /**
+     * Verify if a contact could be a valid contact in its phone
+     *
+     * @param contactPhone  Contact phone to verify
+     *
+     * @return true if contact is valid, false otherwise
+     */
+    public boolean isValidContactPhone(@NonNull String contactPhone){
+        try {
+            new SMSPeer(contactPhone);
+        }catch (InvalidTelephoneNumberException addressException){
+            return false;
+        }
+        return true;
+    }
+
+    //---------------------------- OPERATIONS ON THE CONTACTS DATABASE ----------------------------
+
+    /**
+     * Add a {@link SMSPeer} as {@link SMSContact} in {@link SMSContactDatabase}
+     * after using {@link SMSContactConverterUtils} to convert SMSPeer in a Contact entity
+     *
+     * @param peer      {@link SMSPeer} to insert in the contacts database
+     */
+    public void addContact(@NonNull SMSPeer peer) {
+        SMSContact newContact = SMSContactConverterUtils.contactFromPeer(peer);
+        contactDatabase.access().insert(newContact);
+    }
+
+    /**
+     * Add a {@link SMSPeer} as {@link SMSContact} in {@link SMSContactDatabase}
+     * after using {@link SMSContactConverterUtils} to convert SMSPeer in a Contact entity
+     *
+     * @param peer        {@link SMSPeer} to insert in the contacts database
+     * @param nameContact Optional name for the new contact
+     */
+    public void addContact(@NonNull SMSPeer peer, @NonNull String nameContact) {
+        SMSContact newContact = SMSContactConverterUtils.contactFromPeer(peer, nameContact);
+        contactDatabase.access().insert(newContact);
```

Comment:
Same here, you can use method nesting:
```suggestion
        addContact(newContact);
```